syft/frameworks/torch/tensors/interpreters/gradients_core.py

```diff
@@ -38,6 +46,70 @@ def __call__(self, grad):
     def __repr__(self):
         return self.__class__.__name__
 
+    def __setattr__(self, name, value):
+        self.__dict__[name] = value
+
+        # add attributes of grad function to _parameters list
+        # essential for serialization and deserialization
+        if not name in ["next_functions", "result", "_parameters"]:
+            self._parameters.append(value)
+
+    @staticmethod
+    def simplify(worker: AbstractWorker, grad_fn) -> tuple:
+        """ Takes the attributes of a grad_fn object and saves them in a tuple
+            Every gradient function class that extends `GradFunc` uses this function to
+            simplify the attributes.
+
+        Args:
+            grad_fn: gradient function object (AddBackward, SubBackward, etc.)
+
+        Returns:
+            grad_fn_params: a tuple containing all the simplified attributes
+            of gradient function
+        """
+
+        # Add class name of the grad_fn object at the begining.
+        # Essential while deserializing the object
+        cls = syft.serde.msgpack.serde._simplify(worker, grad_fn.__class__.__name__)
+        grad_fn_params = [cls]
+
+        for param in grad_fn._parameters:
+
+            if hasattr(param, "child"):
+                param = syft.serde.msgpack.serde._simplify(worker, param.child)
```

Comment:
Why do you transmit param.child and not param?

Reply:
My intuition was here we need to push every element of the chain in the tuple (like we do AutogradTensor). But that will be taken care of.
Hence the code is changed and now transmitting param instead of param.child.